### PR TITLE
Enabled golangci-lint with staticcheck only and fixed issues

### PR DESCRIPTION
**Pull request by andresuribe87**
This PR configures and enables golangci-lint in github actions. I intentionally only enabled the `staticcheck` linter, so this becomes easier to review. The only noteworthy change to highlight was the due to the following warning: 

```
pkg/server/router/dwn.go:82:19: SA4022: the address of a variable cannot be nil (staticcheck)
        if err != nil || &publishManifestResponse.Manifest == nil {
```

To fix, I changed this to use the `IsEmpty()` method.

Additional linters will be enabled in a follow up PR.

### CODE REVIEW

File: pkg/server/router/dwn.go [hunk 1]
```diff
@@ -46,8 +46,8 @@ func (req PublishManifestRequest) ToServiceRequest() dwn.PublishManifestRequest
 }
 
 type PublishManifestResponse struct {
-	Manifest    manifest.CredentialManifest       `json:"manifest" validate:"required"`
-	DWNResponse dwnpkg.DWNPublishManifestResponse `json:"dwnResponse" validate:"required"`
+	Manifest    *manifest.CredentialManifest       `json:"manifest" validate:"required"`
+	DWNResponse *dwnpkg.DWNPublishManifestResponse `json:"dwnResponse" validate:"required"`
```
Comment (decentralgabe) [inline pkg/server/router/dwn.go:50]: i'd prefer reverting this, required fields should not be nil-able

File: pkg/server/router/dwn.go [hunk 2]
```diff
@@ -79,20 +79,20 @@ func (dwnr DWNRouter) PublishManifest(ctx context.Context, w http.ResponseWriter
 	req := request.ToServiceRequest()
 	publishManifestResponse, err := dwnr.service.GetManifest(req)
 
-	if err != nil || &publishManifestResponse.Manifest == nil {
+	if err != nil || publishManifestResponse.Manifest == nil {
```
Comment (decentralgabe) [inline pkg/server/router/dwn.go:82]: to get around this check we should define an `IsEmpty` method on Manifest. I think one may even already exist.

**Review by decentralgabe — CHANGES_REQUESTED** [review]

File: pkg/server/router/dwn.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/server/router/dwn.go]: Done.

File: pkg/server/router/dwn.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/server/router/dwn.go]: Sounds good.

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**